test/src/main/java/org/apache/accumulo/test/functional/TabletServerResourceGroupBalanceIT.java

```diff
@@ -0,0 +1,291 @@
+
+      AtomicReference<Exception> error = new AtomicReference<>();
+      Thread ingest = new Thread(() -> {
+        try {
+          ReadWriteIT.ingest(client, 1000, 1, 1, 0, tableName);
+          ReadWriteIT.verify(client, 1000, 1, 1, 0, tableName);
+        } catch (Exception e) {
+          error.set(e);
+        }
+      });
+      ingest.start();
+
+      assertEquals(0, getCountOfHostedTablets(client, tableName));
+
+      // Start TabletServer for GROUP2
+      getCluster().getConfig().getClusterServerConfiguration()
+          .addTabletServerResourceGroup("GROUP2", 1);
+      getCluster().getClusterControl().start(ServerType.TABLET_SERVER);
+
+      client.instanceOperations().waitForBalance();
+      assertEquals(26, getCountOfHostedTablets(client, tableName));
+      ingest.join();
+      assertNull(error.get());
+
+      client.tableOperations().delete(tableName);
+      // Stop all tablet servers because there is no way to just stop
+      // the GROUP2 server yet.
+      getCluster().getClusterControl().stopAllServers(ServerType.TABLET_SERVER);
+      getCluster().getConfig().getClusterServerConfiguration().clearTServerResourceGroups();
+      getCluster().getConfig().getClusterServerConfiguration()
+          .addTabletServerResourceGroup("GROUP1", 1);
+      getCluster().getClusterControl().start(ServerType.TABLET_SERVER);
+
+    }
+  }
+
+  @Test
+  public void testResourceGroupPropertyChange() throws Exception {
+
+    SortedSet<Text> splits = new TreeSet<>();
+    IntStream.range(97, 122).forEach(i -> splits.add(new Text(new String("" + i))));
+
+    NewTableConfiguration ntc1 = new NewTableConfiguration();
+    ntc1.withInitialHostingGoal(TabletHostingGoal.ALWAYS);
+    ntc1.withSplits(splits);
+
+    String tableName = this.getUniqueNames(1)[0];
+    try (final AccumuloClient client =
+        Accumulo.newClient().from(getCluster().getClientProperties()).build()) {
+
+      client.tableOperations().create(tableName, ntc1);
+      client.instanceOperations().waitForBalance();
+
+      assertEquals(26, getCountOfHostedTablets(client, tableName));
+
+      Map<String,String> tserverGroups = getTServerGroups();
+      assertEquals(2, tserverGroups.size());
+
+      Ample ample = ((ClientContext) client).getAmple();
+      String tableId = client.tableOperations().tableIdMap().get(tableName);
+      List<TabletMetadata> locations = ample.readTablets().forTable(TableId.of(tableId))
+          .fetch(TabletMetadata.ColumnType.LOCATION).build().stream().collect(Collectors.toList());
+      assertEquals(26, locations.size());
+      Location l1 = locations.get(0).getLocation();
+      assertEquals("default", tserverGroups.get(l1.getHostAndPort().toString()));
+      locations.forEach(loc -> assertEquals(l1, loc.getLocation()));
+
+      client.tableOperations().setProperty(tableName, "table.custom.assignment.group", "GROUP1");
+      // Wait 2x the MAC default Manager TGW interval
+      Thread.sleep(10_000);
```

Comment:
Instead of sleeping for a fixed time, could refactor the test to wait for things to be in the expected groups.  May make the test a bit more stable.

Reply:
Removed sleep in d0b0c4d